File: app/src/main/java/com/porkandlager/travelonbudget/wires/models/acta/Acta.java
package com.porkandlager.travelonbudget.wires.models.acta;

import com.google.gson.annotations.SerializedName;

/**
 * Created by tista on 11/12/16.
 */

public class Acta<T> {

    @SerializedName("actor")
    private ActaObj actor;

    @SerializedName("action")
    private String action;

    @SerializedName("object")
    private ActaObj object;

    @SerializedName("meta")
    private T meta;

    public T getMeta() {
        return meta;
    }

    public String getAction() {
        return action;
    }

    public void setAction(String action) {
        this.action = action;
    }

    public ActaObj getObject() {
        return object;
    }

    public void setObject(ActaObj object) {
        this.object = object;
    }

    public void setMeta(T meta) {
        this.meta = meta;
    }

    public ActaObj getActor() {
        return actor;
    }

    public void setActor(ActaObj actor) {
        this.actor = actor;
    }
}

File: app/src/main/java/com/porkandlager/travelonbudget/wires/models/acta/Acta.kt
package com.porkandlager.travelonbudget.wires.models.acta

import com.google.gson.annotations.SerializedName

/**
 * Created by tista on 11/12/16.
 */

class Acta<T> {

    @SerializedName("actor")
    var actor: ActaObj? = null

    @SerializedName("action")
    var action: String? = null

    @SerializedName("object")
    var `object`: ActaObj? = null

    @SerializedName("meta")
    var meta: T? = null

}
